### PR TITLE
Adds view transition animations for blog previews

### PR DESCRIPTION
**Pull request by martrapp**
Hi @isabelroses 👋🏼 

As discussed, here's a suggestion for how the blog previews could be animated to feel a bit more lively.
Please find the CSS rules in base.css. 

I also added some blocking code to media.css to pair the existing prefers-reduced-motion styles.

### CODE REVIEW

File: styles/partials/blog-preview.scss [hunk 1]
```diff
@@ -15,6 +15,7 @@
   border: var(--border, var(--default-border));
   display: flex;
   position: relative;
+  background-color: var(--bg);
```
Comment (martrapp) [inline styles/partials/blog-preview.scss:18]: This is to remove the transparency from the original card for more solid view transitions

File: templates/layouts/base.tera [hunk 2]
```diff
@@ -14,6 +14,7 @@
     <meta name="description" content="{{ description }}" />
     <meta name="keywords" content="Isabel Roses, blog, go, rust, personal, tech, programming, web development, software engineering" />
     <link rel="canonical" href="https://isabelrose.com">
+    <link rel=expect href="#below-the-fold" blocking=render>
```
Comment (martrapp) [inline templates/layouts/base.tera:17]: This one ensures that view transitions do not start to early. I the fragment identifier is not defined, the page loads completely before the view transition starts. For a long page, you can define this like `<a name="below-the-fold"/>` somewhere below the fold.

**Review by isabelroses — APPROVED** [review]
Thanks martin your the best :)

**Comment by martrapp** [comment]
Thank you! It was a lot of fun and very insightful. I’m always on the lookout for good real-life examples for https://vtbag.dev. Have fun with the examples! You can see knobs and levers you can turn and pull ;-)

Ping me if you'd like help morphing the previews into the blog articles or other view transition related stuff on your site.

**Comment by isabelroses** [comment]
> Thank you! It was a lot of fun and very insightful. I’m always on the lookout for good real-life examples for [vtbag.dev](https://vtbag.dev). Have fun with the examples! You can see the knobs and levers you can turn and pull ;-)
> 
> Ping me if you'd like help morphing the previews into the blog articles or other view transition related stuff on your site.

Thank you. I'll be sure to keep an eye on your work.

**Pull request merged**